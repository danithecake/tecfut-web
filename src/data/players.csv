name,firstname,lastname,alias,photo_url
dangol,Данил,Головачев,Данил,
antsav,Антон,Савельев,Антон,
kolya,Коля,,Коля,
kosura,Костя,Ураков,Костя,
lesha,Леша,,Леша,
vitya,Витя,,Витя,
zhechi,Женя,Чистов,Женя,
vitaku,Виталик,Акульшин,Виталик Акульшин,
andrey,Андрей,,Андрей,
andbor,Андрей,Борода,Андрей Борода,
ivan,Иван,,Иван,
maxche,Максим,Чернов,Максим,
vitvol,Виталик,Волнушкин,Виталик Волнушкин,
dimsha,Дима,Шальнев,Дима Шальнев,
alerom,Саша,Романихин,Саша Романихин,
mismar,Миша,Меренков,Миша,
