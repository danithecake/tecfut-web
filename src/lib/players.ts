import type { IPlayer } from 'common/interfaces'
import fs from 'fs'
import path from 'path'
import Papa from 'papaparse'

const DATA_DIR = path.join(process.cwd(), 'src', 'data')
const PAPA_CONF: Papa.ParseConfig<IPlayer> = {
  header: true,
  dynamicTyping: true,
  skipEmptyLines: true,
  transformHeader: (header) => (header === 'photo_url' ? 'photoURL' : header)
}

type PlayerName = IPlayer['name']

export const getPlayers: () => Record<PlayerName, IPlayer> = () => {
  try {
    return Papa.parse(fs.readFileSync(path.join(DATA_DIR, 'players.csv'), 'utf8'), PAPA_CONF).data.reduce(
      (players, player) => {
        players[player.name] = player

        return players
      },
      {} as Record<PlayerName, IPlayer>
    )
  } catch (err) {
    console.error(err)

    return {}
  }
}
