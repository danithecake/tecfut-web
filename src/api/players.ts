import type { EndpointOutput } from '@sveltejs/kit'
import { getPlayers } from 'lib/players'

export const get: () => Promise<EndpointOutput> = () => {
  return new Promise((resolve) => void resolve({ body: JSON.stringify(getPlayers()) }))
}
