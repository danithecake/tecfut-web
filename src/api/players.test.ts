import { get } from './players'

describe('api/players', () => {
  describe('get', () => {
    test('returns parsed players map from local players.csv file', async () => {
      expect(JSON.parse((await get()).body as string)).toMatchSnapshot()
    })
  })
})
