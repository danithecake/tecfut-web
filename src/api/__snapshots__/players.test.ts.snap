// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/players get returns parsed players map from local players.csv file 1`] = `
Object {
  "alerom": Object {
    "alias": "Саша Романихин",
    "firstname": "Саша",
    "lastname": "Романихин",
    "name": "alerom",
    "photoURL": null,
  },
  "andbor": Object {
    "alias": "Андрей Борода",
    "firstname": "Андрей",
    "lastname": "Борода",
    "name": "andbor",
    "photoURL": null,
  },
  "andrey": Object {
    "alias": "Андрей",
    "firstname": "Андрей",
    "lastname": null,
    "name": "andrey",
    "photoURL": null,
  },
  "antsav": Object {
    "alias": "Антон",
    "firstname": "Антон",
    "lastname": "Савельев",
    "name": "antsav",
    "photoURL": null,
  },
  "dangol": Object {
    "alias": "Данил",
    "firstname": "Данил",
    "lastname": "Головачев",
    "name": "dangol",
    "photoURL": null,
  },
  "dimsha": Object {
    "alias": "Дима Шальнев",
    "firstname": "Дима",
    "lastname": "Шальнев",
    "name": "dimsha",
    "photoURL": null,
  },
  "ivan": Object {
    "alias": "Иван",
    "firstname": "Иван",
    "lastname": null,
    "name": "ivan",
    "photoURL": null,
  },
  "kolya": Object {
    "alias": "Коля",
    "firstname": "Коля",
    "lastname": null,
    "name": "kolya",
    "photoURL": null,
  },
  "kosura": Object {
    "alias": "Костя",
    "firstname": "Костя",
    "lastname": "Ураков",
    "name": "kosura",
    "photoURL": null,
  },
  "lesha": Object {
    "alias": "Леша",
    "firstname": "Леша",
    "lastname": null,
    "name": "lesha",
    "photoURL": null,
  },
  "maxche": Object {
    "alias": "Максим",
    "firstname": "Максим",
    "lastname": "Чернов",
    "name": "maxche",
    "photoURL": null,
  },
  "mismar": Object {
    "alias": "Миша",
    "firstname": "Миша",
    "lastname": "Меренков",
    "name": "mismar",
    "photoURL": null,
  },
  "vitaku": Object {
    "alias": "Виталик Акульшин",
    "firstname": "Виталик",
    "lastname": "Акульшин",
    "name": "vitaku",
    "photoURL": null,
  },
  "vitvol": Object {
    "alias": "Виталик Волнушкин",
    "firstname": "Виталик",
    "lastname": "Волнушкин",
    "name": "vitvol",
    "photoURL": null,
  },
  "vitya": Object {
    "alias": "Витя",
    "firstname": "Витя",
    "lastname": null,
    "name": "vitya",
    "photoURL": null,
  },
  "zhechi": Object {
    "alias": "Женя",
    "firstname": "Женя",
    "lastname": "Чистов",
    "name": "zhechi",
    "photoURL": null,
  },
}
`;
